ng2-material/components/paging/paging.ts

```diff
@@ -0,0 +1,286 @@
+    result = result.replace(/\{end\}/gi, rangeStop.toString());
+    result = result.replace(/\{total\}/gi, totalItems.toString());
+
+    return result;
+  }
+
+  /**
+   * calculate range depending via model parameters
+   *
+   * @param {IPagingModel} model
+   * @private
+   */
+  private getRange() {
+    if (isPresent(this.model)) {
+      let _rangeStart = (this.model.currentPage - 1) * this.model.itemsPerPage + 1;
+
+      let rest = this.model.totalItems - _rangeStart,
+        _rangeStop = rest < this.model.itemsPerPage ? this.model.totalItems : _rangeStart + this.model.itemsPerPage - 1;
+
+      return this._getFormattedValue(_rangeStart, _rangeStop, this.model.totalItems);
+    }
+
+    return;
+  }
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.paging;
+      });
+  }
+}
+
+@Component({
+  selector: 'md-paging-controls',
+  template: `
+      <span [class.md-paging-control-active]="!isFirstPage()" class="md-paging-control md-paging-control-previous">
+        <button (click)="previousPage()" class="material-icons">keyboard_arrow_left</button>
+      </span>
+      <span [class.md-paging-control-active]="!isLastPage()" class="md-paging-control md-paging-control-next">
+        <button (click)="nextPage()" class="material-icons">keyboard_arrow_right</button>
+      </span>
+    `,
+  host: {
+    '[class.md-paging-controls]': 'true'
+  }
+})
+export class MdPagingControls {
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPagingModel;
+
+  isFirstPage() {
+    return isPresent(this.model) && this.model.currentPage == 1;
+  }
+
+  isLastPage() {
+    return isPresent(this.model) && this.model.totalItems <= this.model.currentPage * this.model.itemsPerPage;
+  }
+
+  previousPage() {
+    this._changePage(this.model.currentPage - 1);
+  }
+
+  nextPage() {
+    this._changePage(this.model.currentPage + 1);
+  }
+
+  _changePage(newPage: number) {
+    let _model = JSON.parse(JSON.stringify(this.model));
+    _model.currentPage = newPage;
+    this._service.change(this.name, _model);
+  }
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.paging;
+      });
+  }
+}
+
+@Component({
+  selector: 'md-paging-length-selector',
+  template: `
+   <span *ngIf="selectorLengthBefore" class="md-paging-length-selector-label md-paging-length-selector-before">{{selectorLengthBefore}}</span>
+   <select [(ngModel)]="model.itemsPerPage" (ngModelChange)="changePagingLength($event)" class="md-paging-length-select">
+      <option *ngFor="#length of selectorLengthAvailable" [value]="length">
```

Comment:
Could be something like `pageLenghChoices` or `pageLengthOptions` - unless I misunderstand, these are options for what the page size could be, correct?